modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/util/ObjectValidationRuleThreadLocal.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.validation.rule.util;
+
+import com.liferay.petra.lang.CentralizedThreadLocal;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author Carolina Barbosa
+ */
+public class ObjectValidationRuleThreadLocal {
+
+	public static void addObjectEntryId(
+		long objectEntryId, long objectValidationRuleId) {
+
+		Map<Long, Set<Long>> objectEntryIdsMap =
+			_objectEntryIdsMapThreadLocal.get();
+
+		Set<Long> objectEntryIds = objectEntryIdsMap.computeIfAbsent(
+			objectValidationRuleId, key -> new HashSet<>());
+
+		objectEntryIds.add(objectEntryId);
+	}
+
+	public static boolean isSkipObjectValidationRuleExecution(
```

Comment:
Why the logic of this method is different from [ObjectActionThreadLocal.isSkipObjectActionExecution](https://github.com/liferay/liferay-portal/blob/e15f84ee11abac54ee86da392eb0c8bcc6442401/modules/apps/object/object-api/src/main/java/com/liferay/object/action/util/ObjectActionThreadLocal.java#L56-L58)? Carol said that she copied and pasted this code from ObjectActionThreadLocal class